File: src/test/kotlin/trustline/appuser/repository/VerificationRepository.java
package trustline.appuser.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import trustline.appuser.model.VerificationModel;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface VerificationRepository extends JpaRepository<VerificationModel, UUID> {
    Optional<VerificationModel> findByUserIdAndPin(UUID userId, String pin);
}

File: src/main/kotlin/trustline/appuser/repository/VerificationRepository.kt
package trustline.appuser.repository;

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import trustline.appuser.dto.Status
import trustline.notification.model.VerificationModel
import java.util.*

@Repository
interface VerificationRepository : JpaRepository<VerificationModel, UUID> {
    fun findByUserIdAndPin(userId: UUID, pin: String): VerificationModel?
    fun findByUserIdAndPinAndStatus(userId: UUID, pin: String, status: Status): VerificationModel?
    fun findByIdAndStatus(id: UUID, status: Status): VerificationModel?
}
